Rename network authentication and encryption variables for configuration.

JoshLind opened the pull request:
## Motivation

This renames two variables in the network configuration to avoid confusion. These variable renaming changes are: (i) 'is_permissioned' -> 'enable_remote_authentication'; and (ii) 'enable_encryption_and_authentication' -> 'enable_noise'. 

(The old variable names were somewhat misleading!)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Local runs of 'cargo xtest' and libra-swarm seem to produce the expected output.

## Related PRs

None.

davidiw commented on config/src/config/network_config.rs:
```diff
@@ -35,17 +35,17 @@ pub struct NetworkConfig {
-    // node can connect. If this flag is set to true, the `enable_encryption_and_authentication`
+    pub enable_noise: bool,
+    // If network authentication is enabled, only trusted peers are allowed to connect. Otherwise, any
+    // node can connect. If this flag is set to true, the `enable_noise`
```
the should be removed before `enable_noise`

JoshLind replied:
Done, thanks!

davidiw commented on config/src/config/network_config.rs:
```diff
@@ -140,10 +140,10 @@ impl NetworkConfig {
                 "Validators should not define network_peers"
             );
-        } else if self.is_permissioned {
+        } else if self.enable_remote_authentication {
```
there's actually no way to enforce ids outside of the other peer enforcing remote authentication, we should consider deleting this piece of code...  we should consider deleting this code. cc: @phlip9

JoshLind replied:
As discussed offline, we'll address these logic changes in a future PR. This PR is just for renaming.

davidiw commented on config/src/config/network_config.rs:
```diff
@@ -140,10 +140,10 @@ impl NetworkConfig {
                 "Validators should not define network_peers"
             );
-        } else if self.is_permissioned {
+        } else if self.enable_remote_authentication {
```
please add a check in this function that ensure noise is enabled if enable_remote_authentication is

JoshLind replied:
As discussed offline, we'll address these logic changes in a future PR. This PR is just for renaming.

davidiw commented on libra-node/src/main_node.rs:
```diff
@@ -120,11 +120,11 @@ pub fn setup_network(
+        // authentication and encryption.
         assert!(
-            config.enable_encryption_and_authentication,
+            config.enable_noise,
```
better to assert in the config code so that we can validate configs in the config code only

JoshLind replied:
As discussed offline, we'll address these logic changes in a future PR. This PR is just for renaming.

JoshLind commented:
Thanks @davidiw! It's ready for another look 😄

JoshLind commented:
Thanks @davidiw! It's ready for another look 😄

davidiw commented:
Looks like you have a conflict you need to fix before landing.

JoshLind commented:
@davidiw, thanks for pointing that out! I've rebased from master and fixed the conflicts. If you're happy with the changes, any chance you could run the libra-bors command? Hopefully we get it fixed next week so that I have appropriate permissions!

davidiw commented:
@bors-libra r+

bors-libra commented:
:pushpin: Commit 1e51ce6 has been approved by `davidiw`

<!-- @bors-libra r=davidiw 1e51ce66fff34f591df3b07266557d2b15af6126 -->

bors-libra commented:
:hourglass: Testing commit 1e51ce66fff34f591df3b07266557d2b15af6126 with merge 70c8fdf14175c9d8fd9210d55ceb07f8bb56c38f...

bors-libra commented:
:sunny: Test successful - [checks-circle_commit_workflow](https://circleci.com/workflow-run/fae16e08-5487-4340-bb32-3aedeaf53679?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)
Approved by: davidiw
Pushing 70c8fdf14175c9d8fd9210d55ceb07f8bb56c38f to master...
